### DISCUSSION TITLE
请教一下，关于异步任务的使用问题

### DISCUSSION THREAD

**Discussion by loveyou3thousand** [BYJRK/DotNet-Discussions#176] [C# 技术交流]
问题的初始在于想做一个异步任务队列，希望先进先执行，执行完了以后接着执行下一个；
于是脑补了很多方式，continuewith(nextTask)、去写一个while(concurrentqueue.trydequeue(out var sthRun))然后semaphoreslim(1,1)控制、while(concurrentqueue.trydequeue(out var sthRun))然后wait()、while(concurrentqueue.trydequeue(out var sthRun))然后await task.waitasync()。
但是我不知道哪种是合理的设计方式，分别有什么错误和漏洞，以及task.wait()、await task.waitasync()、await task分别应该怎么使用，对应什么业务需求的时候会立马想到这么用，合理且高效
希望有人可以在闲暇之余指点一下，感谢

**Comment by AIDEBUGTO** [comment]
我粗陋的觉得使用栈的方式

**Comment by BYJRK** [comment]
异步编程可以考虑用 Channel，而不是 ConcurrentQueue
